### PR TITLE
Improve the Environment error reporting APIs

### PR DESCRIPTION
**Pull request by jopit**
#### Description:
- Added a new status condition which will be less subject to misinterpretation

### CODE REVIEW

File: appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go [hunk 1]
```diff
@@ -340,10 +340,98 @@ var _ = Describe("Environment controller tests", func() {
...
+			Expect(env.Status.Conditions[0].Status).To(Equal(metav1.ConditionFalse))
+			Expect(env.Status.Conditions[0].Reason).To(Equal(EnvironmentReasonErrorOccurred + "Resolved"))
+			Expect(env.Status.Conditions[0].Message).To(BeEmpty())
+			Expect(env.Status.Conditions[1].Type).To(Equal(EnvironmentConditionReconciled))
+			Expect(env.Status.Conditions[1].Status).To(Equal(metav1.ConditionTrue))
+			Expect(env.Status.Conditions[1].Reason).To(Equal(EnvironmentReasonSecretNotFound + "Resolved"))
+			Expect(env.Status.Conditions[1].Message).To(BeEmpty())
+		})
+
 		It("should not return an error if the Environment does not container UnstableConfigurationFields", func() {
```
Comment (Rizwana777) [inline appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go:435]: typo : `container --> contains,` this typo is not from your changes, I think can be fixed in this PR

**Comment by Rizwana777** [comment]
LGTM, left a minor comment for a typo which is not from your changes

File: appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go (hunk 1 above)
Comment (jopit) [reply appstudio-controller/controllers/appstudio.redhat.com/environment_controller_test.go]: Fixed.

**Review by jgwest — APPROVED** [review]
Looks perfect, very nice, thanks @jopit, and thanks @Rizwana777 for reviewing!

**Pull request merged**